lint/shellcheck.bzl

```diff
@@ -62,9 +67,10 @@ def _shellcheck_aspect_impl(target, ctx):
     if ctx.rule.kind not in ["sh_binary", "sh_library"]:
         return []
 
-    patch, report, info = patch_and_report_files(_MNEMONIC, target, ctx)
-    shellcheck_action(ctx, ctx.executable._shellcheck, filter_srcs(ctx.rule), ctx.file._config_file, report, ctx.attr._options[LintOptionsInfo].fail_on_violation)
```

Comment:
what are `LintOptionsInfo` used for? can the `fail_on_violation` attribute be removed with this change?

Reply:
No, we still need it - but the logic to access it was moved to the shared helper function in `lint/private/lint_aspect.bzl`. That's nice because now access is in the same file as the definition.

Reply:
Another way to explain: we want actions to succeed, it's then up to whatever tool interprets the lint results to decide whether developers workflow should be halted.